### ISSUE TITLE
[Cms] Accessing nested options works inconsistent

### ISSUE THREAD

**Issue by lukaskleinschmidt** [getkirby/kirby#1515]
**Describe the bug**
Trying to get nested options with keys containing dots (e.g. plugin options) does not work as expected.

```php
Kirby::plugin('namespace/plugin', [
    'options' => [
        'key' => 'A',
        'nested' => [
          'key' => 'B'
        ]
    ]
]);

// available options
$options = [
    'namespace.plugin.key' => 'A',
    'namespace.plugin.nested' => [
        'key' => 'B',
    ],
];

option('namespace.plugin.key') // will return A
option('namespace.plugin.nested.key') // will return null expected B
option('namespace.plugin.nested')['key'] // will return B
```

**Additional context**
I actually wanted to create a ticket addressing some problems on how options are stored. But after almost creating the issue I found that it already was addressed some time ago.
So this issue is now only about the problem of getting nested options from plugins. Although the issue could resolve itself when transforming the dot notation consequently to a multidimensional array like already addressed in this [ideas issue](https://github.com/getkirby/ideas/issues/32) and all its related issues.

<details>
<summary>The initial issue</summary>
This is perhaps more of an feature than a bug. But I'm not quite sure so here it goes.

For my config folder I like it to be as clean and readable as possible. Therefore I try to seperate several configs to appropiate files and include them in the main config. The one thing that prevents me from doing this for plugins as well is following issue.

```php
Kirby::plugin('namespace/plugin', [
    'options' => [
        'key' => 'A',
    ]
]);

// results in
$options = [
    'namespace.plugin.key' => 'A'
];
```

This makes it impossible to outsource configs for a specific plugin in one file.

```php
// config/namespace.plugin.php
return [
    'key' => 'B',
];

// config/config.php
return [
    'namespace.plugin' => require_once __DIR__ . '/namespace.plugin.php',
];

// results in
$options = [
    'namespace.plugin.key' => 'A',
    'namespace.plugin' => [
        'key' => 'B'
    ]
];
```

I know that I can easily work around this by outsourcing each individual option of a plugin to a seperate file. But this feels kinda cumbersome and not that elegant.

I also noticed that if I have nested options that `A::get()` breaks for keys containing dots.

```php
Kirby::plugin('namespace/plugin', [
    'options' => [
        'key' => 'A',
        'nested' => [
          'key' => 'C'
        ]
    ]
]);

option('namespace.plugin.key') // will return A
option('namespace.plugin.nested.key') // will return null
option('namespace.plugin.nested')['key'] // will return C
```

**To sum things up**
I would expect that the `option()` getter works for nested options (as well as described above).
And I expected that plugin options are stored as an array.
```php
Kirby::plugin('namespace/plugin', [
    'options' => [
        'a' => 'A',
        'b' => 'B',
    ]
]);

// expected
$options = [
    'namespace.plugin' => [
        'a' => 'A',
        'b' => 'B',
    ]
];

// current behaviour
$options = [
    'namespace.plugin.a' => 'A',
    'namespace.plugin.b' => 'B',
];
```
</details>

**Comment by bastianallgeier** [comment]
✅
